shame_command.py

```diff
@@ -0,0 +1,80 @@
+import configparser
+import logging
+import os
+import aiohttp
+import discord
+from discord.ext import commands
+
+logger = logging.getLogger(__name__)
+
+# Load the config file
+config_path = os.path.join(os.path.dirname(__file__), "settings.cfg")
+config = configparser.ConfigParser()
+config.read(config_path)
+
+TODOIST_API = "https://api.todoist.com/rest/v2/tasks"
+
+
+# Retrieve Todoist API Key for the email from the config file
+def get_todoist_token(email: str) -> str:
+    config.read(config_path)
+    if email in config["TODOIST_KEY_BY_EMAIL"]:
+        return config["TODOIST_KEY_BY_EMAIL"][email]
+    else:
+        return None
+
+
+# Get tasks from Todoist API with 'shame' label using aiohttp
+async def get_shame_tasks(todoist_token: str, session: aiohttp.ClientSession) -> list:
+    headers = {"Authorization": f"Bearer {todoist_token}"}
+    params = {"filter": "label:shame"}
+
+    async with session.get(TODOIST_API, headers=headers, params=params) as response:
+        if response.status != 200:
+            logger.error(f"Failed to retrieve tasks: {response.status}")
+            return []
```

Comment:
catching is good here, but this feels like a quiet failure mode for the end user- if the call fails it would just show that the user has no tasks

maybe an optional None return value that the caller can catch would work? or even just throwing an exception and catching it higher in the chain (if the task pull fails, the rest of the execution is meaningless)

Reply:
yeah I think this is a good point, since a client response error seems outside the normal operation, lets throw an exception here